etl/steps/data/garden/unesco/2024-06-16/education_opri.py

```diff
@@ -0,0 +1,112 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("education_opri")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["education_opri"].reset_index()
+
+    # Retrieve snapshot with the metadata provided via World Bank.
+
+    snap_wb = paths.load_snapshot("edstats_metadata.xls")
+    tb_wb = snap_wb.read()
+
+    #
+    # Process data.
+    #
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+    tb_pivoted = tb.pivot(index=["country", "year"], columns="indicator_label_en", values="value")
+
+    # Add metadata columns to the table
+    long_definition = add_metadata_wb_edstats(tb, tb_wb)
+    tb_pivoted = add_metadata_description(tb_pivoted, long_definition)
+
+    tb_pivoted = tb_pivoted.reset_index()
+    tb_pivoted = tb_pivoted.format(["country", "year"])
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb_pivoted], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def add_metadata_wb_edstats(tb, tb_wb):
+    long_definition = {}
+    for indicator in tb["indicator_label_en"].unique():
+        long_definition[indicator] = tb_wb[tb_wb["Indicator Name"] == indicator]["Long definition"].values
+    return long_definition
+
+
+def add_metadata_description(tb, long_definition):
+    for column in tb.columns:
+        meta = tb[column].metadata
+
+        # Use the function
+        meta.description_from_producer = long_definition[column]
+        meta.title = column
+        meta.processing = "minor"
+        meta.display = {}
+
+        # Convert the 'name' variable to lowercase to make text matching easier
+        name_lower = tb[column].title.lower()
+
+        #
+        # Update metadata units, short_units and number of decimal places to display depending on what keywords the variable name contains
+        #
+        # Fill out units and decimal places
+        if "%" in name_lower:
```

Comment:
I don't think you should change this, but you could check out Jinja templates as an alternative (an example [here](https://github.com/owid/etl/blob/master/etl/steps/data/garden/ihme_gbd/2024-05-20/gbd_cause.meta.yml)), I find them a bit more readable than adding metadata in the script, but up to you!

Reply:
thank you! I've tried it but I think it will be tricky to add the long descriptions to the metadata using Jinja templates. looks like everything that we might want to use to define metadata in Jinja needs to be a part of the index that then gets pivoted in the grapher stage? maybe I am missing something though..